plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -1,23 +1,145 @@
 package com.google.cloud.tools.eclipse.appengine.localserver.server;
 
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.ui.console.MessageConsoleStream;
+import org.eclipse.wst.server.core.IModule;
 import org.eclipse.wst.server.core.IServer;
+import org.eclipse.wst.server.core.model.IModuleResource;
+import org.eclipse.wst.server.core.model.IModuleResourceDelta;
 import org.eclipse.wst.server.core.model.ServerBehaviourDelegate;
 
+import com.google.cloud.tools.appengine.api.AppEngineException;
+import com.google.cloud.tools.appengine.api.devserver.AppEngineDevServer;
+import com.google.cloud.tools.appengine.api.devserver.DefaultRunConfiguration;
+import com.google.cloud.tools.appengine.api.devserver.DefaultStopConfiguration;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkAppEngineDevServer;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessExitListener;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessStartListener;
+
+/**
+ * A {@link ServerBehaviourDelegate} for App Engine Server executed via the Java App Management
+ * Client Library.
+ */
 public class LocalAppEngineServerBehaviour extends ServerBehaviourDelegate {
+  private LocalAppEngineStartListener localAppEngineStartListener;
+  private LocalAppEngineExitListener localAppEngineExitListener;
+  private AppEngineDevServer devServer;
+
+  public LocalAppEngineServerBehaviour () {
+	localAppEngineStartListener = new LocalAppEngineStartListener();
+	localAppEngineExitListener = new LocalAppEngineExitListener();
+  }
 
   @Override
   public void stop(boolean force) {
     setServerState(IServer.STATE_STOPPING);
-    // TODO stop server
+    // stop server
+    terminate();
     setServerState(IServer.STATE_STOPPED);    
   }
 
-  void setStarted() {
-    setServerState(IServer.STATE_STARTED);    
+  /**
+   * Convenience method allowing access to protected method in superclass.
+   */
+  @Override
+  protected IModuleResourceDelta[] getPublishedResourceDelta(IModule[] module) {
+    return super.getPublishedResourceDelta(module);
   }
-  
-  void setStarting() {
-    setServerState(IServer.STATE_STARTING);    
+
+  /**
+   * Convenience method allowing access to protected method in superclass.
+   */
+  @Override
+  protected IModuleResource[] getResources(IModule[] module) {
```

Comment:
The method getPublishedResourceDelta(IModule[]) and getResources(IModule[]) from the type ServerBehaviourDelegate is not visible in LocalAppEnginePublishOperation.java